app/models.py

```diff
@@ -1386,6 +1387,14 @@ class Notification(db.Model):
         {},
     )
 
+    @cached_property
```

Comment:
I added this as a cached property, to further prevent unneeded db queries.  I doubt it's really needed - can switch to a regular property if you'd prefer.

This is a quality of life improvement so you don't have to drill deep in to determine the "default send" of a notification